get_goes_x.py

```diff
@@ -11,99 +11,148 @@
-    try:
-        table = h5.root.data
-        lasttime = table.col('time')[-1]
-        ok = newdat['time'] > lasttime
-        newdat = newdat[ok]
-        h5.root.data.append(newdat)
-    except tables.NoSuchNodeError:
-        table = h5.create_table(h5.root, 'data', newdat,
-                               "GOES_X rates", expectedrows=2e7)
-    h5.root.data.flush()
-
+        print(('Malformed GOES_X data so table read failed: {}'
+               .format(err)))
+        sys.exit(0)
+    return dat
+
+
+def process_xray_data(dat, satellite):
+    """
+    Take the astropy table of 'dat' and rearrange to give a single
+    row for each sample time, include the historical time
+    columns, and add a column to describe the satellite.
+    """
+
+    # Select only the GOES satellite specified from the args
+    dat = dat[dat['satellite'] == satellite]
+    if len(dat) == 0:
+        print('Warning: No data in fetched file for satellite {}'.format(satellite))
+        sys.exit(0)
+
+    # Make a table for each of the two wavelengths
+    shortdat = dat[dat['energy'] == '0.05-0.4nm']['flux', 'time_tag']
+    shortdat.rename_column('flux', 'short')
+    longdat = dat[dat['energy'] == '0.1-0.8nm']['flux', 'time_tag']
+    longdat.rename_column('flux', 'long')
+    if len(longdat) != len(shortdat):
+        print('Warning: "short" and "long" table have mismatched lengths')
+
+    # Join them on time (seems to be OK for these data)
+    joindat = join(shortdat, longdat)
+
+    # Add a time column with Chandra secs and remove original time_tags
+    times = Time(joindat['time_tag'])
+    joindat['time'] = times.cxcsec
+    joindat.remove_column('time_tag')
+
+    # Add the other columns the old file format wanted
+    joindat['mjd'] = times.mjd.astype(int)
+    joindat['secs'] = np.array(np.round((times.mjd - joindat['mjd']) * 86400,
+                                        decimals=0)).astype(int)
+    joindat['year'] = [t.year for t in times.datetime]
+    joindat['month'] = [t.month for t in times.datetime]
+    joindat['dom'] = [t.day for t in times.datetime]
+    joindat['hhmm'] = np.array([
+        f"{t.hour}{t.minute:02}" for t in times.datetime]).astype(int)
+
+    joindat['ratio'] = -100000.0
+    ok = (joindat['long'] != 0) & (joindat['long'] != -100000.0)
+    joindat['ratio'][ok] = joindat['short'][ok] / joindat['long'][ok]
+
+    # Return table for h5 file
+    joindat['satellite'] = satellite
+    return joindat['year', 'month', 'dom', 'hhmm', 'mjd', 'secs',
+                   'short', 'long', 'ratio', 'time', 'satellite'].as_array()
+
+
+def main():
+
+    args = get_options()
+
+    # Read the data file just to get the last record
+    with tables.open_file(args.h5, mode='r',
+                          filters=tables.Filters(complevel=5, complib='zlib')) as h5:
+        try:
+            table = h5.root.data
+            lasttime = table.col('time')[-1]
+        except:
```

Comment:
You should be specific with exceptions where possible.  In this case there is a specific exception `NoSuchNodeError` that should be caught.  Other exceptions should be raised.

Reply:
I follow your reasoning in general on exception handling,  but I also couldn't figure out a case (specific to arc) where we'd want the job to fail due to explicit exception handling instead of continuing.